### ISSUE TITLE
Add checks to confirm that the checkpoint conversion script works perfectly correct

### ISSUE THREAD

**Issue by ibeltagy** [bigscience-workshop/Megatron-DeepSpeed#114]
We now have a script that convert megatron-deepspeed checkpoints to HF-transformers checkpoints.
Project is [here](https://github.com/bigscience-workshop/Megatron-DeepSpeed/tree/main/tools/convert_checkpoint) and the script is [here](https://github.com/bigscience-workshop/Megatron-DeepSpeed/blob/main/tools/convert_checkpoint/deepspeed_to_transformers.py). 
However, the script doesn't have unit tests that confirm that the conversion is correct. 

The goal of this issue is to add such tests. The idea is to run the forward pass of both models (before and after conversion) with any random input, then use `torch.allclose` to assert that the output loss and the logits of both models perfectly match.

**Comment by ibeltagy** [comment]
[Here](https://huggingface.co/bigscience/gpt2-350m-en/tree/megatron-deepspeed)'s a megatron-deepspeed checkpoint and [here](https://huggingface.co/bigscience/gpt2-350m-en/tree/main)'s the corresponding HF-transformer checkpoint. We just need to verify that these two are the same.

**Comment by StellaAthena** [comment]
git clone https://huggingface.co/bigscience/gpt2-350m-en/tree/megatron-deepspeed is failing. It says repo not found. I can download the HF version easily.

**Comment by ibeltagy** [comment]
@stas00?

**Comment by stas00** [comment]
The correct syntax is:
```
git clone --single-branch --branch megatron-deepspeed https://huggingface.co/bigscience/gpt2-350m-en
```

**Comment by jaketae** [comment]
@StellaAthena Have you made progress with this issue? If not, I'd like to take a jab at it. 

@stas00 Would the unit test run with the CI? I'm wondering if/whether the test script would have to download the Megatron checkpoints manually on each run.

**Comment by StellaAthena** [comment]
> @StellaAthena Have you made progress with this issue? If not, perhaps I'll take a jab at it!
> 
> @stas00 Will the unit test run with the CI? I'm wondering if/whether the test script would have to download the Megatron checkpoints manually on each run.

I have not been able to get to this, ICLR stuff has been getting in the way. You’re welcome to take it over.

**Comment by stas00** [comment]
> @stas00 Will the unit test run with the CI? I'm wondering if/whether the test script would have to download the Megatron checkpoints manually on each run.

The AWS-based CI is currently borked, need to start from scratch and build on GCS, so we do manual `make test` for now.

Re your question - there is no need to use on a huge checkpoint because both of the download and it'd be much more difficult to compare. It should be easy to create a tiny checkpoint of a few mbs on the fly and then convert it and then compare.

Let me know if you run into difficulties with that.

**Comment by thomasw21** [comment]
Reposting from slack as this seems relevant for this:

>I'm looking at transformers GPT2 code. https://huggingface.co/transformers/_modules/transformers/models/gpt2/modeling_gpt2.html#GPT2Model and it seems it is doing post layernom whereas the 13B one is trained using PreLN. Maybe this is why we're seeing poor performance in evaluation? 
Typically the number of params is the same, just the way we use them is different. Is there a preLN gpt in transformers?
